doc/lua_api.txt

```diff
@@ -2434,8 +2434,11 @@ These functions return the leftover itemstack.
     * Optional: Variable number of arguments that are passed to `func`
 
 ### Server
-* `minetest.request_shutdown([message],[reconnect])`: request for server shutdown. Will display `message` to clients,
-    and `reconnect` == true displays a reconnect button.
+* `minetest.request_shutdown([message],[reconnect],[delay])`: request for server shutdown. Will display `message` to clients,
+    `reconnect` == true displays a reconnect button,
+    `delay` permit to add an optionnal delay before shutdown
+        negative delay cancel the current active shutdown
```

Comment:
i think this is very very overengineered for a shutdown. you should only have core or one mod at maximum to do this action on your server else its the chaos on your shutdown behaviour